index.js

```diff
@@ -0,0 +1,295 @@
+const reportStatus = (message, type) => {
+  if (type) {
+    $('#status-message').html(message).attr('class', type);
+  } else {
+    $('#status-message').html(message).attr('class', '');
+  }
+};
+
+const reportError = (message, errors) => {
+  let content = `<span>${message}</span>`;
+  content += '<div>';
+
+  for (const field in errors ){
+    for (const problem of errors[field]) {
+      content += `<span>${field}: ${problem}&emsp;</span>`;
+    }
+  }
+  content += '</div>';
+  reportStatus(content, 'failure');
+};
+
+const baseURL = 'https://ada-backtrek-api.herokuapp.com/trips';
+
+const buildTripTable = function buildTripTable(input) {
+
+  $('#trip-list').append('<tbody>');
+
+  input.data.forEach( (trip) => {
+    let content = '<tr>';
+    content += `<td id=${trip.id}>${trip.name}</td>`;
+    content += `</tr>`;
+    $('#trip-list').append(content);
+  });
+
+  $('#trip-list').append('</tbody>');
+  reportStatus(`${input.data.length} trips loaded`, 'success');
+}
+
+const getTrips = () => {
+  const tripList = $('#trip-list');
+  tripList.empty();
+
+  reportStatus('Loading trips...')
+
+  axios.get(baseURL)
+    .then((response) => {
+      $('#trip-list').empty();
+      $('#trip-list').append('<h3>All Trips<h3>');
+      buildTripTable(response);
```

Comment:
Yes this is awesome! I like that the you saw the opportunity to DRY up this code when adding some of the optionals